src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheckTest.java

```diff
@@ -174,4 +174,50 @@ public void testClearState() throws Exception {
                 .isTrue();
     }
 
+    /**
+     * Tries to reproduce system failure to call Check on not acceptable token.
+     * It can not be reproduced by Input files. Maintainers thinks that keeping
+     * exception on unknown token is beneficial.
+     *
+     */
+    @Test
+    public void testImproperTokenOnLeaveToken() {
+        final ReturnCountCheck check = new ReturnCountCheck();
+
+        final DetailAstImpl classDefAst = new DetailAstImpl();
+        classDefAst.setType(TokenTypes.CLASS_DEF);
+
+        try {
+            check.leaveToken(classDefAst);
+            assertWithMessage("IllegalStateException is expected").fail();
+        }
+        catch (IllegalStateException exception) {
+            assertWithMessage("Message doesn't contain ast")
+                    .that(exception.getMessage())
+                    .isEqualTo(classDefAst.toString());
+        }
```

Comment:
Can we use junit5 capability to test exceptions? E.g.:
```
final IllegalStateException exception = assertThrows(IllegalStateException.class,
        () -> check.leaveToken(classDefAst), "IllegalStateException was expected");

assertWithMessage("Message doesn't contain ast")
        .that(exception.getMessage())
        .isEqualTo(classDefAst.toString());
```

Reply:
@rdiachenko please check i have done it